code/datums/spells/greaterknock.dm

```diff
@@ -0,0 +1,42 @@
+/obj/effect/proc_holder/spell/aoe_turf/greaterknock
+	name = "Greater Knock"
+	desc = "On first cast, will remove access restrictions on all airlocks on the Cyberiad, and announce this spell's use and your arrival to the station. On any further cast, will open all doors in sight. Cannot be refunded once bought!"
+
+	school = "transmutation"
+	charge_max = 200
+	clothes_req = 0
+	invocation = "MAIOR OXIN FIERA"
+	invocation_type = "shout"
+	range = 7
+	cooldown_min = 40 //40 deciseconds reduction per rank
+	action_icon_state = "knock"
+	sound = 'sound/magic/Knock.ogg'
+	var/used = FALSE
+
+/obj/effect/proc_holder/spell/aoe_turf/greaterknock/cast(list/targets, mob/user = usr)
+	if (!used)
+		for(var/obj/machinery/door/airlock/A in airlocks)
+			if(is_station_level(A.z))
+				A.req_access = list()
+		if (is_station_level(usr.z))
```

Comment:
I wouldn't bother with this check, just check for used and do the rest regardless of the wizard's z-level

Also, same as above regarding spacing

Reply:
Wizard z-level check is purely for fluff, it changes the text on the announcement. The spell works either way